toss/scripts/scaling_performance_benchmark.py

```diff
@@ -0,0 +1,142 @@
+import sys
+sys.path.append('..')
+
+# Core packages
+from math import pi
+import numpy as np
+import time
+import pygmo as pg
+
+
+def load_udp(args, lower_bounds, upper_bounds, number_of_islands, population_size, number_of_generations):
+    """
+    Main function for optimizing the initial state for deterministic trajectories around a 
+    small celestial body using a mesh.
+    """
+    # Load required modules
+    from udp_initial_condition import udp_initial_condition
+
+    start_time = time.time()
+
+    # Setup User-Defined Problem (UDP)
+    print("Setting up the UDP...")
+    udp = udp_initial_condition(args, lower_bounds, upper_bounds)
+    prob = pg.problem(udp)
+
+    # Setup User-Defined Algorithm (UDA)
+    print("Setting up UDA")
+    uda = pg.gaco(
+        number_of_generations, 
+        args.algorithm.kernel_size, 
+        args.algorithm.convergence_speed_parameter, 
+        args.algorithm.oracle_parameter, 
+        args.algorithm.accuracy_parameter, 
+        args.algorithm.threshold_parameter, 
+        args.algorithm.std_convergence_speed_parameter, 
+        args.algorithm.improvement_stopping_criterion, 
+        args.algorithm.evaluation_stopping_criterion, 
+        args.algorithm.focus_parameter, 
+        args.algorithm.memory_parameter)
+    default_bfe = pg.bfe() # Default batch-fitness-evaluation for parallellized initialization of archipelago.
```

Comment:
Don't you have to use one of the parallelized BFEs, e.g. https://esa.github.io/pygmo2/bfes.html#pygmo.thread_bfe ?

See also https://esa.github.io/pygmo2/overview.html#list-of-batch-fitness-evaluators